### ISSUE TITLE
Transition will run twice onStart when the animation begin？

### ISSUE THREAD

**Issue by JQChan** [drcmda/react-spring#244]
I found Transition ran twice onStart when the animation began, But I found that the example in the API docs run only one time ,here is my code: 
``` javascript
 render() {
    const { className, style, active, onStart, onDestroyed } = this.props;
    return (
      <Transition
        keys={String(active)}
        onStart={onStart}
        onDestroyed={onDestroyed}
        from={{ opacity: 0, height: 0 }}
        enter={{ opacity: 1, height: 'auto' }}
        leave={{ opacity: 0, height: 0 }}
        config={config.molasses}>
        {
          active && ((styles) =>
            <animated.div style={styles}>
              <div className={classNames('br-collapse', className)} style={style}>
                {this.props.children}
              </div>
            </animated.div>)
        }
      </Transition>
    );
  }
```
Can someone help me? Is that something wrong in my Code?

**Comment by drcmda** [comment]
I see it, will have to look into it more deeply 🤔

Just asking, for what do you need onStart? Because onStart in transition is a little arbitrary since it is a stack, it will call onStart on elements about to be destroyed as well or for updates and entering elements. In other words, are you sure you want a transition? To me it looks like some sort of expander, in that case it should be a spring, which is simpler to handle and faster since react doesn't mount/unmount contents.

**Comment by JQChan** [comment]
Ah, it looks like expander. I  thought about that the expander could  call onStart when it opened and call onDestroy when it fold use by  Transition when I saw the docs: (since spring will call twice onStart and onRest)
<p>If you need to toggle a single child, that is also possible.</p>

``` javascript
<Transition
  from={{ position: 'absolute', opacity: 0 }}
  enter={{ opacity: 1 }} 
  leave={{ opacity: 0 }}>
  {visible && (styles => <div style={styles}>Single Component</div>)}
</Transition>
```

never mind, I think you are right, maybe spring is a good resolve way

**Comment by drcmda** [comment]
The upside is that a spring is faster since react doesn't have to remount. It's sometimes good to keep stuff you want to show around, even when it's height:0, display:none. I am trying to track down the bug in the meantime, but if you want you can put the full expander source into your box, maybe there's a better way because usually relying on onRest/Start isn't so optimal.

**Comment by raine** [comment]
> your transition uses animated.div but there's no native specified on the transition. More about that here: http://react-spring.surge.sh/native

Is `native` a thing with `useTransition`? I haven't found any mention of it in the hooks api documentation.

**Comment by drcmda** [comment]
no, it's native by default. the "native" flag doesn't exist any longer. "null" isn't nothing in react, it's a valid component in the render-tree. the transition doesn't know your view, it doesn't you will print "null" out to the dom, it just runs a enter/leave/update.
